R/particle_filter.R

```diff
@@ -109,9 +111,11 @@ particle_filter <- R6::R6Class(
     ##' @param save_history Logical, indicating if the history of all
     ##' particles should be saved
     ##'
+    ##' @param user Optional user parameters to use when creating the model
+    ##'
     ##' @return A single numeric value representing the log-likelihood
     ##' (\code{-Inf} if the model is impossible)
-    run = function(state, n_particles, save_history = FALSE) {
+    run = function(state, n_particles, save_history = FALSE, user = NULL) {
```

Comment:
Can you overload functions in R? Do you think there would be any use in having a run which also initialises, for cases where the model is only run once? If not, asking for two calls to the object isn't a problem

Reply:
I'm not sure I follow here, where are two calls to the object?